air/src/air/mod.rs

```diff
@@ -184,6 +192,67 @@ pub trait Air: Send + Sync {
+    ///
+    /// The evaluations should be written into the `results` slice in the same order as the
+    /// the order of auxiliary transition constraint degree descriptors used to instantiate
+    /// [AirContext] for this AIR. Thus, the length of the `result` slice will equal to the number
+    /// of auxiliary transition constraints defined for this computation.
+    ///
+    /// The default implementation of this function panics. It must be overridden for AIRs
+    /// describing computations which require multiple trace segments.
+    ///
+    /// The types for main and auxiliary trace evaluation frames are defined as follows:
+    /// * When the entire protocol is executed in a prime field, types `F` and `E` are the same,
+    ///   and thus, both the main and the auxiliary trace frames are defined over the base filed.
+    /// * When the protocol is executed in an extension field, the main trace frame is defined
+    ///   over the base field, while the auxiliary trace frame is defined over the extension field.
+    ///
+    /// We define type `F` separately from `Self::BaseField` to allow evaluation of constraints
+    /// over the out-of-domain evaluation frame, which may be defined over an extension field
+    /// (when extension fields are used). The type bounds specified for this function allow the
+    /// following:
+    /// * `F` and `E` could be the same [StarkField] or extensions of the same [StarkField].
+    /// * `F` and `E` could be the same field, because a field is always an extension of itself.
+    /// * If `F` and `E` are different, then `E` must be an extension of `F`.
+    #[allow(unused_variables)]
+    fn evaluate_aux_transition<F, E>(
+        &self,
+        main_frame: &EvaluationFrame<F>,
+        aux_frame: &EvaluationFrame<E>,
+        periodic_values: &[F],
+        aux_rand_elements: &AuxTraceRandElements<E>,
+        result: &mut [E],
+    ) -> Vec<Assertion<F>>
+    where
+        F: FieldElement<BaseField = Self::BaseField>,
+        E: FieldElement<BaseField = Self::BaseField> + ExtensionOf<F>,
+    {
+        unimplemented!("evaluation of auxiliary transition constraints has not been implemented");
+    }
+
+    /// Returns a set of assertions placed against auxiliary trace segments.
+    ///
+    /// When the protocol is executed using an extension field, auxiliary assertions are defined
+    /// over the extension field. This is in contrast with the assertions returned from
+    /// [get_assertions()](Air::get_assertions) function, which always returns assertions defined
+    /// over the base field of the protocol.
+    ///
+    /// The default implementation of this function returns an empty vector. It should be
+    /// overridden only if the computation relies on auxiliary trace segments. In such a case,
+    /// the vector returned from this function must contain at least one assertion.
+    #[allow(unused_variables)]
+    fn get_aux_assertions<E: FieldElement<BaseField = Self::BaseField>>(
```

Comment:
> RAP columns will (always?) be start = end = 1, on each auxiliary column

In most cases yes, but there could be a few important cases when that may not be so:

1. For initializing "public memory" (like described in Cairo paper) the starting value would not be 1.
2. In some cases, it may be desirable to keep the running product of a table (basically like an accumulator) without removing anything from that able. In such a case the end value would not be 1.

Reply:
Ah good point, I was not thinking about these usecases. Probably better to keep it that way then!